test/likelihood.jl

```diff
@@ -10,100 +10,58 @@ tspan = (0.0,10.0)
 diff_distributions = [fit_mle(Normal,aggregate_data[i,j,:]-aggregate_data[i,j-1,:]) for j in 2:200, i in 1:2 ]
-
 obj = build_loss_objective(prob1,Tsit5(),LogLikeLoss(t,data_distributions,diff_distributions),
                                      maxiters=10000,verbose=false)
-
 bound1 = Tuple{Float64, Float64}[(0.5, 5),(0.5, 5)]
 result = bboptimize(obj;SearchRange = bound1, MaxSteps = 11e3)
-
 @test result.archive_output.best_candidate ≈ [1.5,1.0] atol = 1e-1
 
-pf_func = function (du,u,p,t)
-  du[1] = p[1] * u[1] - p[2] * u[1]*u[2]
-  du[2] = -3.0 * u[2] + u[1]*u[2]
-end
-u0 = [1.0;1.0]
-tspan = (0.0,10.0)
-p = [1.5,1.0]
-prob1 = ODEProblem(pf_func,u0,tspan,p)
-t = collect(linspace(0,10,200))
-sol = solve(prob1,Tsit5(),saveat=t)
-
-function generate_data(sol,t)
-  randomized = VectorOfArray([(sol(t[i]) + .01randn(2)) for i in 1:length(t)])
-  data = convert(Array,randomized)
-end
-aggregate_data = convert(Array,VectorOfArray([generate_data(sol,t) for i in 1:100])) 
+ 
 data_distributions = [fit_mle(Normal,aggregate_data[i,j,:]) for i in 1:2, j in 1:200]
 diff_distributions = [fit_mle(Normal,aggregate_data[i,j,:]-aggregate_data[i,j-1,:]) for j in 2:200, i in 1:2 ]
-
 obj = build_loss_objective(prob1,Tsit5(),LogLikeLoss(t,data_distributions,diff_distributions,0.3),
                                      maxiters=10000,verbose=false)
-
 bound1 = Tuple{Float64, Float64}[(0.5, 5),(0.5, 5)]
 result = bboptimize(obj;SearchRange = bound1, MaxSteps = 11e3)
-
 @test result.archive_output.best_candidate ≈ [1.5,1.0] atol = 1e-1
 
-pf_func = function (du,u,p,t)
-  du[1] = p[1] * u[1] - p[2] * u[1]*u[2]
-  du[2] = -3.0 * u[2] + u[1]*u[2]
-end
-u0 = [1.0;1.0]
-tspan = (0.0,10.0)
-p = [1.5,1.0]
-prob1 = ODEProblem(pf_func,u0,tspan,p)
-sol = solve(prob1,Tsit5())
 
-t = collect(linspace(0,10,200))
-function generate_data(sol,t)
-  randomized = VectorOfArray([(sol(t[i]) + .01randn(2)) for i in 1:length(t)])
-  data = convert(Array,randomized)
-end
-aggregate_data = convert(Array,VectorOfArray([generate_data(sol,t) for i in 1:100]))
 distributions = [fit_mle(MvNormal,aggregate_data[:,j,:]) for j in 1:200]
 diff_distributions = [fit_mle(MvNormal,aggregate_data[:,j,:]-aggregate_data[:,j-1,:]) for j in 2:200]
-
 obj = build_loss_objective(prob1,Tsit5(),LogLikeLoss(t,distributions,diff_distributions),
                                      maxiters=10000,verbose=false)
-
 bound1 = Tuple{Float64, Float64}[(0.5, 5),(0.5, 5)]
 result = bboptimize(obj;SearchRange = bound1, MaxSteps = 11e3)
+@test result.archive_output.best_candidate ≈ [1.5,1.0] atol = 1e-1
+
 
+obj = build_loss_objective(prob1,Tsit5(),L2Loss(t,aggregate_data),
+                                     maxiters=10000,verbose=false)
+bound1 = Tuple{Float64, Float64}[(0.5, 5),(0.5, 5)]
+result = bboptimize(obj;SearchRange = bound1, MaxSteps = 11e3)
 @test result.archive_output.best_candidate ≈ [1.5,1.0] atol = 1e-1
+
+
+aggregate_data_diff = aggregate_data[:,1:end-1,:] - aggregate_data[:,2:end,:] 
```

Comment:
the user shouldn't need to generate this

Reply:
Oh, this is the cause of that warning. `aggregate_data_diff = aggregate_data[:,1:end-1,:] - aggregate_data[:,2:end,:] ` that's a 3D matrix and in the cost function you're indexing it in 2D.